# Case Study 02- Measures of Central Tendency and Dispersion

In [1]:
import pandas as pd

In [2]:
filename='StudentsPerformance.csv'
Stu_df=pd.read_csv(filename)

In [3]:
Stu_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


**It can be observed from above that the student data set has no nulls**

In [6]:
Stu_df['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

# 1)There are 518 females and 482 males in the study

In [7]:
Stu_df['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

# 2)The parents' level of education has five values, and most parents have a degree of some kind.
#     It can be seen that the average value of the parental level of education is "some college" since this variable is at ordinal level (so the mode is used)

# 3)Mean Scores

In [12]:
Stu_df.groupby(Stu_df['gender'])["math score"].mean().to_frame()

math score
gender            
female   63.376448
male     68.821577


# Maths: 
**a)From the above, it can be seen that the mean value of math score is higher for males than females**

In [10]:
Stu_df['test preparation course'].value_counts()

none         642
completed    358
Name: test preparation course, dtype: int64

In [11]:
Stu_df.groupby(Stu_df['test preparation course'])["math score"].mean().to_frame()

math score
test preparation course            
completed                  69.96648
none                       63.78972

**b)From the above, it is seen that 642 students did not attend a preparation while the remaining 358 students attended a preparation session. The math score mean for the group that did the course was 69.97 while it was 63.79 for the group that didn't complete it**

In [13]:
Stu_df.groupby(Stu_df['gender'])["reading score"].mean().to_frame()

reading score
gender               
female      72.590734
male        65.545643

# Reading score:
**From above, it is seen that the mean score for reading is higher for females compared to males**

In [14]:
Stu_df.groupby(Stu_df['test preparation course'])["reading score"].mean().to_frame()

reading score
test preparation course               
completed                    74.175978
none                         66.417445

**b)From above, it is observed that mean reading score is higher for those who did the preparation than for those who did not**

In [15]:
Stu_df.groupby(Stu_df['gender'])["writing score"].mean().to_frame()

writing score
gender               
female      72.467181
male        63.446058

# Writing score:
**a)From above it is observed that writing score is higher for females than for males**

In [16]:
Stu_df.groupby(Stu_df['test preparation course'])["writing score"].mean().to_frame()

writing score
test preparation course               
completed                    74.684358
none                         64.457944

**b) From above it is seen that writing score is significantly higher for those who completed the preparation than for those who did not**

# 4)Variance by Category

In [17]:
Stu_df.var()

math score       242.411410
reading score    216.279254
writing score    231.842682
dtype: float64

# The values of variance overall for the dataset is as shown above

In [18]:
male_filt=Stu_df['gender']=='male'
female_filt=Stu_df['gender']=='female'
prep_yes_filt=Stu_df['test preparation course']=='completed'
prep_no_filt=Stu_df['test preparation course']=='none'


**a)**

**Males , math score**

In [19]:
Stu_df.loc[male_filt, 'math score'].var()

211.88909688494755

**Males, all subjects variance**

In [23]:
Stu_df.loc[male_filt, : ].var()

math score       211.889097
reading score    200.211010
writing score    202.413924
dtype: float64

**Average math score of males who did the prep**

In [21]:
Stu_df.loc[male_filt&prep_yes_filt,'math score'].mean()

72.98850574712644

**b)**

**Females, math score**

In [20]:
Stu_df.loc[female_filt, 'math score'].var()

256.9585931607201

**Females, all subjects variance**

In [24]:
Stu_df.loc[female_filt, : ].var()

math score       256.958593
reading score    207.677438
writing score    220.369327
dtype: float64

**Average math score of Females who did the prep course**

In [22]:
Stu_df.loc[female_filt&prep_yes_filt,'math score'].mean()

67.1086956521739

**c)**

**Variation based on test prep**

In [25]:
Stu_df.loc[prep_yes_filt,:].var()

math score       210.884027
reading score    183.265864
writing score    175.202612
dtype: float64

In [26]:
Stu_df.loc[prep_no_filt,:].var()

math score       246.668662
reading score    213.419851
writing score    226.251739
dtype: float64

**From the above, it can be seen that the variation is less among the students who attended the preparation course than for those who did not attend the prep, for every subject**


# Variation in a single data frame

In [27]:
Stu_df.groupby(['gender','test preparation course'])['math score','reading score','writing score'].var()

<ipython-input-27-ca9bb6a36e6a>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  Stu_df.groupby(['gender','test preparation course'])['math score','reading score','writing score'].var()


math score  reading score  writing score
gender test preparation course                                          
female completed                208.173913     161.175546     149.836897
       none                     272.602767     214.241220     225.495172
male   completed                197.098133     185.297987     166.098233
       none                     205.513600     184.843553     181.394687

# 5)List of students who secured the top 25 percentile on math score

In [28]:
Stu_df['math score'].quantile(0.75)

77.0

In [47]:
Math_Top_Qrt_fil = Stu_df['math score']>=77.0

#Filtering the students who scored more than or equal to 77.0 percent

In [36]:
Stu_df.loc[Math_Top_Qrt_fil, :].count()

gender                         255
race/ethnicity                 255
parental level of education    255
lunch                          255
test preparation course        255
math score                     255
reading score                  255
writing score                  255
dtype: int64

# It can be seen that the number of students eligible for scholarship is 255

In [38]:
Scholarship_df = Stu_df.loc[Math_Top_Qrt_fil, :]

In [41]:
Scholarship_df.reset_index(inplace=True)

In [46]:
Scholarship_df.loc[0:254:14]

index  gender race/ethnicity parental level of education         lunch  \
0      414  female        group A                 high school  free/reduced   
14     428  female        group D                some college  free/reduced   
28     442  female        group E          associate's degree      standard   
42     456  female        group E             master's degree      standard   
56     470  female        group E          associate's degree  free/reduced   
70     484  female        group C                some college      standard   
84     498  female        group C          associate's degree      standard   
98     512  female        group D                some college      standard   
112    857    male        group A                some college      standard   
126    871    male        group C                some college      standard   
140    885    male        group D             master's degree      standard   
154    899    male        group D                some college      standard   
168    913    male        group E            some high school      standard   
182    927    male        group B            some high school      standard   
196    941    male        group B           bachelor's degree  free/reduced   
210    955    male        group D                some college      standard   
224    969    male        group E             master's degree      standard   
238    983    male        group C           bachelor's degree      standard   
252    997    male        group D                some college      standard   

    test preparation course  math score  reading score  writing score  
0                 completed          77             88             85  
14                     none          77             86             86  
28                completed          79             88             94  
42                     none          81             92             91  
56                completed          83             86             88  
70                completed          87             89             94  
84                     none          91             95             94  
98                     none          98            100             99  
112               completed          78             72             70  
126               completed          79             79             78  
140                    none          80             80             72  
154                    none          81             82             84  
168                    none          82             67             61  
182               completed          85             84             78  
196               completed          87             90             88  
210                    none          88             73             78  
224                    none          90             85             84  
238               completed          94             90             91  
252               completed         100             97             99

**The above data frame shows every 14th row in the list of students selected for scholarship, for sample**

# *Thank You*